### DISCUSSION TITLE
It would be nice if we don't get the Integer Division warning when the expression handles it correctly

### DISCUSSION THREAD

**Discussion by stephannv** [godotengine/godot-proposals#10516] [Editor]
eg. 
```gdscript
var a = int(amount / value) # this results in "Integer division, decimal part will be discarded."
```

The analyzer could understand that the integer division is already being handled by the user. 

ps. An existing solution is adding `@warning_ignore("integer_division")` but it would be nice if we don't have to add the annotation or disable the warning.

**Comment by aXu-AP** [comment]
You can actally disable the warning in project settings: `debug/gdscript/warnings/integer_division`

**Reply by stephannv** [reply]
> You can actally disable the warning in project settings: `debug/gdscript/warnings/integer_division`

Yeah, I know, but these warnings are useful, so I don't want to disable it.

**Reply by aXu-AP** [reply]
Ah, sorry, I misunderstood the idea. I'm not sure how the examples you wrote are "correct" handling of the resulting integer - it doesn't need further casting to int and rounding ints with ceil or floor also don't make any sense. It's hard to judge the intention of the user, I think `@warning_ignore` is good for this...

**Reply by stephannv** [reply]
It makes sense. I was using floori and ceili wrong. Thank you for helping me out about this. I updated the description. I think the warning is a helpful way to tell the developer the decimal part is being discarded, so, when I use `int(x / y)` I'm explicitly discarding the decimal part, so it wouldn't be necessary a warning about this or an annotation.
